### ISSUE TITLE
x262 hangs and produces a 0 KB file if intrinsics are turned on

### ISSUE THREAD

**Issue by FranceBB** [kierank/x262#13]
On an Intel Xeon Gold 6238R CPU 56c/112th AVX512 capable, when using all the intrinsics the encode hangs and then eventually crashes.
This is an example:

AVS Script:

```
ColorBars(1920, 1080, pixel_type="YV12")
Converttoyv16()
ChangeFPS(25)
```

BAT File:

`x262_64.exe "D:\Temp\AVS Script.avs" --mpeg2 --preset medium --level high --profile 422 --bitrate 50000 --vbv-maxrate 50000 --vbv-bufsize 17825792 --keyint 12 --bframes 2 --no-scenecut --open-gop --deblock -1:-1 --overscan show --colormatrix bt709 --range tv --transfer bt709 --colorprim bt709 --videoformat component --nal-hrd cbr --output-csp i422 --output "\\mibctvan000\Ingest\MEDIA\temp\raw_video.m2v"`

![image](https://user-images.githubusercontent.com/18946343/139541305-13061e70-c090-4b36-9937-85a18d6212c2.png)

Disabling intrinsics makes it work.

In other words, adding --asm 1 makes it work correctly:

`x262_64.exe "D:\Temp\AVS Script.avs" --mpeg2 --preset medium --level high --profile 422 --bitrate 50000 --vbv-maxrate 50000 --vbv-bufsize 17825792 --keyint 12 --bframes 2 --no-scenecut --open-gop --deblock -1:-1 --overscan show --colormatrix bt709 --range tv --transfer bt709 --colorprim bt709 --videoformat component --nal-hrd cbr --output-csp i422 --asm 1 --output "\\mibctvan000\Ingest\MEDIA\temp\raw_video.m2v"`

![image](https://user-images.githubusercontent.com/18946343/139541310-5e1604db-307b-47bd-ac7f-af73782f31a4.png)

and it goes on normally

![image](https://user-images.githubusercontent.com/18946343/139541316-07d6fe15-5dd9-4b8a-b948-17087613c922.png)

**Comment by ifb** [comment]
Resolved during the rebase onto upstream x264.  Same cause/fix as #7

**Comment by FranceBB** [comment]
Yep, it's working fine now and it's using up to AVX2 (which is the maximum I had on this PC). Thanks! :)
